pkg/nacler/keyers/tpmkeyer/tpmkeyer.go

```diff
@@ -0,0 +1,142 @@
+package tpmkeyer
+
+import (
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/sha256"
+	"errors"
+	"fmt"
+	"io"
+	"runtime"
+	"sync"
+
+	"github.com/google/go-tpm/tpm2"
+	"github.com/google/go-tpm/tpmutil"
+)
+
+type TpmKeyerOption func(*TpmKeyer)
+
+// WithExternalTpm lets the caller provide a tpm to use. This is useful for testing.
+// The caller is responsible for closing the external tpm.
+func WithExternalTpm(externalTpm io.ReadWriteCloser) TpmKeyerOption {
+	return func(t *TpmKeyer) {
+		t.externalTpm = externalTpm
+	}
+}
```

Comment:
Based mostly off this comment, this seems a little weird. Why have a single `TpmKeyer` with internal switching between an internal and external, and not have `InternalTpmKeyer` and `ExternalTpmKeyer` (or whatever)

Reply:
I updated this comment, but I'm not sure if were on the same page. The `TpmKeyer` does not change, what changes is the way we interface with the TPM chip hardware. For example, in the CI environment, we don't have access to a TPM chip, so we set an "ExternalTpm" to the tpm simulator.